spec/services/post_scraper_spec.rb

```diff
@@ -191,101 +195,76 @@
-    stub_fixture(url, 'scrape_no_replies')
-
-    user = create(:user, username: "Marri")
-    board = create(:board, creator: user)
-    nita = create(:character, user: user, screenname: 'wild_pegasus_appeared', name: 'Juanita')
-    icon = create(:icon, keyword: 'sad', url: 'http://v.dreamwidth.org/8517100/2343677', user: user)
-    gallery = create(:gallery, user: user)
-    gallery.icons << icon
-    nita.galleries << gallery
-
-    expect(User.count).to eq(1)
-    expect(Icon.count).to eq(1)
-    expect(Character.count).to eq(1)
-
-    scraper = PostScraper.new(url, board.id)
-    expect(scraper).not_to receive(:print).with("User ID or username for wild_pegasus_appeared? ")
-    expect(scraper.send(:logger)).to receive(:info).with("Importing thread 'linear b'") # just to quiet it
-
-    scraper.scrape!
-    expect(User.count).to eq(1)
-    expect(Icon.count).to eq(1)
-    expect(Character.count).to eq(1)
-  end
-
-  it "doesn't recreate icons if they already exist for that character with new urls" do
-    url = 'http://wild-pegasus-appeared.dreamwidth.org/403.html?style=site&view=flat'
-    stub_fixture(url, 'scrape_no_replies')
+  context "with icons" do
+    let(:user) { create(:user, username: "Marri") }
+    let(:board) { create(:board, creator: user) }
+    let(:icon) { create(:icon, keyword: 'sad', url: 'http://v.dreamwidth.org/8517100/2343677', user: user) }
+    let(:url) { 'http://wild-pegasus-appeared.dreamwidth.org/403.html?style=site&view=flat' }
+    let(:scraper) { PostScraper.new(url, board.id) }
 
-    user = create(:user, username: "Marri")
-    board = create(:board, creator: user)
-    nita = create(:character, user: user, screenname: 'wild_pegasus_appeared', name: 'Juanita')
-    icon = create(:icon, keyword: 'sad', url: 'http://glowfic.com/uploaded/icon.png', user: user)
-    gallery = create(:gallery, user: user)
-    gallery.icons << icon
-    nita.galleries << gallery
+    before(:each) do
+      gallery = create(:gallery, user: user, icons: [icon])
+      create(:character, user: user, screenname: 'wild_pegasus_appeared', name: 'Juanita', galleries: [gallery])
+      stub_fixture(url, 'scrape_no_replies')
+    end
 
-    expect(User.count).to eq(1)
-    expect(Icon.count).to eq(1)
-    expect(Character.count).to eq(1)
+    it "doesn't recreate characters and icons if they exist" do
+      expect(scraper).not_to receive(:print).with("User ID or username for wild_pegasus_appeared? ")
+      expect(scraper.send(:logger)).to receive(:info).with("Importing thread 'linear b'") # just to quiet it
 
-    scraper = PostScraper.new(url, board.id)
-    expect(scraper).not_to receive(:print).with("User ID or username for wild_pegasus_appeared? ")
-    expect(scraper.send(:logger)).to receive(:info).with("Importing thread 'linear b'") # just to quiet it
+      expect { scraper.scrape! }.to not_change{User.count}.and not_change{Icon.count}.and not_change{Character.count}
```

Comment:
This might be more succinct as:

```suggestion
      expect { scraper.scrape! }.to not_change{[User.count, Icon.count, Character.count]}
```